### PR TITLE
Add an issue template, tighten PR template

### PR DESCRIPTION
**Pull request by janetkuo**
Supersedes #28488

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30167)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+manner.  If we can't reproduce a bug or think a feature already exists, we
+might close your issue.  If we're wrong, PLEASE feel free to reopen it and
+explain why.
+-->
+
+----------
+BUG REPORT
+----------
+
+*Kubernetes version*: (try `kubctl version`)
```
Comment (Kashomon) [inline .github/ISSUE_TEMPLATE.md]: Drive-by nit: s/kubctl/kubectl

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+*How to reproduce it, as minimally and precisely as possible*:
+
+
+*Anything else do we need to know*:
+
+
+
+[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/.github/ISSUE_TEMPLATE.md?pixel)]()
```
Comment (Kashomon) [inline .github/ISSUE_TEMPLATE.md]: This analytics tracking pixel thing seems misplaced, no? Should it exist in the issue template?

**Comment by janetkuo** [comment]
**Is this a request for help** (If your answer is yes, you should instead post it in StackOverflow, Slack channel, or mailing list, see http://kubernetes.io/docs/troubleshooting/.):


**What keywords did you search in Kubernetes issues** (If you find any duplicates, you should instead reply there or use the :+1: icon in the upper-right of the issue message):

---

**Is this a BUG REPORT or FEATURE REQUEST** (choose one):



**Kubernetes version** (use `kubectl version`):


**Environment**:
- **Cloud provider**:
- **OS**:
- **kernel**:
- **Install tools**:
- **Others**:


**What happened**:


**What you expected to happen**:


**How to reproduce it, as minimally and precisely as possible**:


**Anything else do we need to know**:



[Analytics]()

**Comment by janetkuo** [comment]
The above is the current issue template

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (janetkuo) [inline .github/ISSUE_TEMPLATE.md]: fixed

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (janetkuo) [inline .github/ISSUE_TEMPLATE.md]: https://github.com/kubernetes/kubernetes/pull/28488#discussion_r70337901 explains why it's here, but the problem is solved now, removed analytics

**Comment by janetkuo** [comment]
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
* Enter your extended release note here (newlines are formatted as bullets).  If this PR does not warrant a release
  note, just say `NONE`.
-OR-
* Use the release-note-* labels to set the release note state (if you have access)
* Clear this block to use the PR title as the release note
```

**Comment by janetkuo** [comment]
The above is the current issue & PR template, major changes include:

- Addressed all comments in previous PR #28488
- Make it shorter (otherwise no one is going to read it)
- Removed `PLEASE REMOVE THIS COMMENT BLOCK BEFORE SUBMITTING` (it won't display anyways so better save this space)
- Made all "read this and that paragraph" more Q&A-ish (so people can't ignore it)
- Issue template: removed developer guide https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md, I don't think people can send a PR without first reading it, and the space in the template is expensive (this information is fighting with other more important information), so removed it

**Comment by janetkuo** [comment]
@bgrant0607 ptal

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->
+
+**Is this a request for help** (If yes, you should instead post it in StackOverflow, Slack channel, or mailing list, see http://kubernetes.io/docs/troubleshooting/.):
```
Comment (philips) [inline .github/ISSUE_TEMPLATE.md]: Perhaps something like: **Is this a request for help** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.)

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+**What keywords did you search in Kubernetes issues before filing this one** (If you found any duplicates, you should instead reply there or use the :+1: icon in the upper-right of the issue message):
```
Comment (philips) [inline .github/ISSUE_TEMPLATE.md]: Missing question mark after "one?".

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+might close your issue.  If we're wrong, PLEASE feel free to reopen it and
+explain why.
+-->
+
+**Kubernetes version** (use `kubectl version`):
+
+
+**Environment**:
+- **Cloud provider**:
+- **OS**:
```
Comment (philips) [inline .github/ISSUE_TEMPLATE.md]: OS (e.g. from /etc/os-release)

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+manner.  If we can't reproduce a bug or think a feature already exists, we
+might close your issue.  If we're wrong, PLEASE feel free to reopen it and
+explain why.
+-->
+
+**Kubernetes version** (use `kubectl version`):
+
+
+**Environment**:
+- **Cloud provider**:
```
Comment (philips) [inline .github/ISSUE_TEMPLATE.md]: Cloud provider or hardware configuration

File: .github/ISSUE_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- **OS**:
+- **kernel**:
```
Comment (philips) [inline .github/ISSUE_TEMPLATE.md]: Kernel (e.g. uname -a)

**Comment by philips** [comment]
Overall this LGTM.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,19 +1,23 @@
...
 -->
 
+**What this PR does / why we need it**:
+
+**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
+
+**Special notes for your reviewer**:
+
+**Release note**:
 ```release-note
```
Comment (bgrant0607) [inline .github/PULL_REQUEST_TEMPLATE.md]: It would be nice if we could make the syntax clear enough to automatically set the release-note labels based on the text here.

**Comment by janetkuo** [comment]
All comments addressed, @philips @bgrant0607 ptal

**Comment by philips** [comment]
LGTM

**Comment by janetkuo** [comment]
Squashed; applying tag
